src/opencmw_worker/src/TimeDomainWorker.hpp

```diff
@@ -179,9 +185,68 @@ class TimeDomainWorker
             sink->set_callback(std::bind(&TimeDomainWorker::callbackCopySinkData, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3, std::placeholders::_4, std::placeholders::_5));
         }
 
-        super_t::setCallback([this](RequestContext & /*rawCtx*/, const TimeDomainContext & /*requestContext*/, const Empty &,
-                                     TimeDomainContext & /*replyContext*/, Acquisition &out) {
-            out = _reply;
+        super_t::setCallback([this](RequestContext & /*rawCtx*/, const TimeDomainContext &requestContext, const Empty &,
+                                     TimeDomainContext & /*replyContext*/, Acquisition   &out) {
+            std::set<std::string, std::less<>> requestedSignals;
+            if (!checkRequestedSignals(requestContext, requestedSignals)) {
+                // break; // TODO: Throw EXCEPTION
+            }
+
+            // find how many chunks should be parallely polled
+            std::vector<uint64_t> chunksAvailable;
+            for (const auto &requestedSignal : requestedSignals) {
+                auto     signalData = _signalsMap.at(requestedSignal);
+                uint64_t diff       = signalData.ringBuffer->cursor() - signalData.eventPoller->sequence()->value();
+                chunksAvailable.push_back(diff);
+            }
+
+            uint64_t maxChunksToPoll = *std::min_element(chunksAvailable.begin(), chunksAvailable.end());
+
+            if (maxChunksToPoll == 0) {
+                // no new items for requested signals
```

Comment:
(comments in this file will probably be addressed in worker related PRs)
Blocking here should reduce latency with the cost of blocking one worker thread per client/subscription, making it a long-polling scheme instead of regular polling.